### DISCUSSION TITLE
Looking for Dockerfile(s) for containers

### DISCUSSION THREAD

**Discussion by brandonhon** [Budibase/budibase#3556] [Help] [answered]
I need to add a few packages to the bbapps container and would like to use the Dockerfile that is used to build the container specified in docker-compose.yml. I think it may be located in the repo under /budibase/packages/ but not 100% sure.

I need python for filling a pdf from data returned in a row under automate.

**Comment by mjashanks** [comment, marked as answer]
Hi Brandon.

There are 2 docker files 
- `packages/worker` - the automations runner.
- `packages/server` - web server.

Please share your forked repository!

**Reply by brandonhon** [reply]
Thanks! I figured that was them just wanted to double check.

**Reply by brandonhon** [reply]
@mjashanks with the below dockerfile that I copied from packages/server I am getting this error after building and trying to run. Any suggestions? I can see the `builder` directory in the image pulled from `budibase.docker.scarf.sh/budibase/apps` but the dockerfile don't seem to create that.

`{"message":"ENOENT: no such file or directory, stat '/app/builder/index.html'","status":404}`

Note: server.tar.gz contains everything in the packages/server directory

```
FROM node:14-buster-slim

RUN apt-get update && apt-get upgrade -y

LABEL com.centurylinklabs.watchtower.lifecycle.pre-check="scripts/watchtower-hooks/pre-check.sh"
LABEL com.centurylinklabs.watchtower.lifecycle.pre-update="scripts/watchtower-hooks/pre-update.sh"
LABEL com.centurylinklabs.watchtower.lifecycle.post-update="scripts/watchtower-hooks/post-update.sh"
LABEL com.centurylinklabs.watchtower.lifecycle.post-check="scripts/watchtower-hooks/post-check.sh"

WORKDIR /app

ENV PORT=4001
ENV COUCH_DB_URL=https://couchdb.budi.live:5984
ENV BUDIBASE_ENVIRONMENT=PRODUCTION

# copy files and install dependencies
# COPY . ./
ADD server.tar.gz ./
RUN yarn
RUN yarn build

# Install client for oracle datasource
RUN apt-get install -y python3 python3-pip git pdftk curl vim unzip libaio1
RUN /bin/bash -e scripts/integrations/oracle/instantclient/linux/x86-64/install.sh
RUN pip3 install pdfrw pdfforms pyexcel
RUN mkdir -p /mnt/data/forms
ADD https://www.atf.gov/firearms/docs/4473-part-1-firearms-transaction-record-over-counter-atf-form-53009/download /mnt/data/forms/4473.pdf

RUN cd /mnt/data/forms && pdfforms inspect -p filled/ 4473.pdf

EXPOSE 4001

# have to add node environment production after install
# due to this causing yarn to stop installing dev dependencies
# which are actually needed to get this environment up and running
ENV NODE_ENV=production
CMD ["yarn", "run:docker"]
```
